.github/workflows/cut-release.yml

```diff
@@ -0,0 +1,59 @@
+name: "cut-release"
+
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'version'
+        required: false
+jobs:
+  cut_release:
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout code 
+        uses: actions/checkout@v3
+      - name: "Build & test"
+        run: |
+          sudo apt-get install build-essential -y
+          ./autogen.sh
+          ./configure
+          make distcheck
+      - name: Update VERSION file on checkout 
+        run: |
+          set -x
+          sudo apt-get install -y python3-semver
+          if [ -z "$NEW_VERSION" ]; then
+             echo "Current Version: $(cat VERSION)"
+             pysemver bump patch $(cat VERSION) > VERSION
+          else
+             echo $NEW_VERSION > VERSION
+          fi
+          export NEW_VERSION=$(cat VERSION)
+          echo "::set-output name=NEW_VERSION::$NEW_VERSION" 
+        id: bump-version-file  
+        env:
+          NEW_VERSION: ${{ inputs.version }}
+      - name: Commit VERSION file 
+        uses: EndBug/add-and-commit@v9
+        with:
+          add: 'VERSION'
+          message: bump semantic version
+          committer_name: GitHub Actions
+          committer_email: actions@github.com
+      - name: Create Tag
+        uses: rickstaa/action-create-tag@v1
+        with:
+          tag: "v${{ steps.bump-version-file.outputs.NEW_VERSION }}"
+          message: "v{{ steps.bump-version-file.outputs.NEW_VERSION }} release"
+      - name: Create Release
+        id: create_release
+        uses: marvinpinto/action-automatic-releases@latest
+        with:
+          repo_token: "${{ secrets.GITHUB_TOKEN }}"
+          automatic_release_tag: v${{ steps.bump-version-file.outputs.NEW_VERSION }}
```

Comment:
Hmmm, isn't the "Create Tag" above redundant if we're going to have the release auto-create the tag?

Reply:
you are right it is not needed I will take it out.

Reply:
So I've been looking at this and it seems this is creating an unsigned tag (see `git help tag` for the `-s` option).

The git "signed" tags are usually better, they create an extra tag object and not just a "pointer" to a commit.

See the difference here:

```
$ git show v2.0.13
tag v2.0.13
Tagger: Filipe Brandenburger <filbranden@gmail.com>
Date:   Fri Aug 30 01:16:08 2019 -0700

Release numactl 2.0.13

commit 5d9f16722e3df49dc618a9f361bd482559695db7 (tag: v2.0.13)
Author: Filipe Brandenburger <filbranden@gmail.com>
Date:   Fri Aug 30 01:11:16 2019 -0700

    Release numactl 2.0.13

    Signed-off-by: Filipe Brandenburger <filbranden@gmail.com>

diff --git a/configure.ac b/configure.ac
index 572b53d..b615f1f 100644
--- a/configure.ac
+++ b/configure.ac
@@ -1,5 +1,5 @@
 AC_PREREQ([2.64])
-AC_INIT([numactl], [2.0.12])
+AC_INIT([numactl], [2.0.13])
```

And:

```
$ git show v2.0.16
commit 7d4f68bbc6e1634e036ef69c45a47902f013545a (HEAD -> main, tag: v2.0.16, lrfurtado/master, lrfurtado/cutrelease, lrfurtado)
Author: Luciano Furtado <lrfurtado@yahoo.com.br>
Date:   Sat Sep 24 17:06:13 2022 +0000

    add cut-release github workflow

diff --git a/.github/workflows/cut-release.yml b/.github/workflows/cut-release.yml
new file mode 100644
index 0000000..277df58
--- /dev/null
+++ b/.github/workflows/cut-release.yml
@@ -0,0 +1,53 @@
+name: "cut-release"
...
```

Ignoring that the tag is pointing at the wrong commit, you can also see it's not a signed tag. Ideally we should use one. Perhaps the separate "Create Tag" action would be able to provide that correctly?

Reply:
Looking at `EndBug/add-and-commit`, it seems it supports [tagging the created commit](https://github.com/EndBug/add-and-commit#tagging), and it seems you can pass it arguments to the `git tag` command that might include the `-a` flag (for an annotated tag) including a `-m` for a message for the annotated tag itself. Can you take a look at that?